<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
<title>OMNeT++ Simulation Library: cRuntimeError Class Reference</title>
<link href="doxygen.css" rel="stylesheet" type="text/css">
<link href="tabs.css" rel="stylesheet" type="text/css">
</head><body>
<!-- Generated by Doxygen 1.5.5 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="dirs.html"><span>Directories</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
</div>
<div class="contents">
<h1>cRuntimeError Class Reference<br>
<small>
[<a class="el" href="group__Internals.html">Internal classes</a>]</small>
</h1><!-- doxytag: class="cRuntimeError" --><!-- doxytag: inherits="cException" --><code>#include &lt;<a class="el" href="cexception_8h-source.html">cexception.h</a>&gt;</code>
<p>
<div class="dynheader">
Inheritance diagram for cRuntimeError:</div>
<div class="dynsection">

<p><center><img src="classcRuntimeError.png" usemap="#cRuntimeError_map" border="0" alt=""></center>
<map name="cRuntimeError_map">
<area href="classcException.html" alt="cException" shape="rect" coords="0,0,93,24">
</map>
</div>

<p>
<a href="classcRuntimeError-members.html">List of all members.</a><hr><a name="_details"></a><h2>Detailed Description</h2>
Thrown when the simulation kernel or other components detect a runtime error. 
<p>
For example, <a class="el" href="classcSimpleModule.html#29a64f1f0c560ac75494940d5c7a048d" title="Schedules a self-message.">cSimpleModule::scheduleAt()</a> throws this exception when the specified simulation time is in the past, or the message pointer is NULL. <table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcRuntimeError.html#acd498d12133cb96ba3dbcac557f91e2">cRuntimeError</a> (OppErrorCode errcode,...)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcRuntimeError.html#8090149405fd25ff4b8e84e360bfb3b4">cRuntimeError</a> (const char *msg,...)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcRuntimeError.html#ed86d404f6019d7b1a9bad020bfa25e7">cRuntimeError</a> (const <a class="el" href="classcObject.html">cObject</a> *where, OppErrorCode errcode,...)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcRuntimeError.html#a7735ad35e64f13776ed3c756b28787e">cRuntimeError</a> (const <a class="el" href="classcObject.html">cObject</a> *where, const char *msg,...)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcRuntimeError.html#413646c577852fa82a848ab0187c57c8">cRuntimeError</a> (const <a class="el" href="classcRuntimeError.html">cRuntimeError</a> &amp;e)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual <a class="el" href="classcRuntimeError.html">cRuntimeError</a> *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classcRuntimeError.html#6d34e0bf66e83e7c209bb72df3b816f8">dup</a> () const </td></tr>

</table>
<hr><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" name="acd498d12133cb96ba3dbcac557f91e2"></a><!-- doxytag: member="cRuntimeError::cRuntimeError" ref="acd498d12133cb96ba3dbcac557f91e2" args="(OppErrorCode errcode,...)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cRuntimeError::cRuntimeError           </td>
          <td>(</td>
          <td class="paramtype">OppErrorCode&nbsp;</td>
          <td class="paramname"> <em>errcode</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&nbsp;</td>
          <td class="paramname"> <em>...</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Error is identified by an error code, and the message comes from a string table. 
<p>
The error string may expect printf-like arguments (s, d) which also have to be passed to the constructor. 
</div>
</div><p>
<a class="anchor" name="8090149405fd25ff4b8e84e360bfb3b4"></a><!-- doxytag: member="cRuntimeError::cRuntimeError" ref="8090149405fd25ff4b8e84e360bfb3b4" args="(const char *msg,...)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cRuntimeError::cRuntimeError           </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&nbsp;</td>
          <td class="paramname"> <em>...</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
To be called like printf(). 
<p>
The error code is set to eCUSTOM. 
</div>
</div><p>
<a class="anchor" name="ed86d404f6019d7b1a9bad020bfa25e7"></a><!-- doxytag: member="cRuntimeError::cRuntimeError" ref="ed86d404f6019d7b1a9bad020bfa25e7" args="(const cObject *where, OppErrorCode errcode,...)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cRuntimeError::cRuntimeError           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classcObject.html">cObject</a> *&nbsp;</td>
          <td class="paramname"> <em>where</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">OppErrorCode&nbsp;</td>
          <td class="paramname"> <em>errcode</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&nbsp;</td>
          <td class="paramname"> <em>...</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Error is identified by an error code, and the message comes from a string table. 
<p>
The error string may expect printf-like arguments (s, d) which also have to be passed to the constructor. The 1st arg is the object where the error occurred: its class and object name will be prepended to the message like this: "(cArray)arr". 
</div>
</div><p>
<a class="anchor" name="a7735ad35e64f13776ed3c756b28787e"></a><!-- doxytag: member="cRuntimeError::cRuntimeError" ref="a7735ad35e64f13776ed3c756b28787e" args="(const cObject *where, const char *msg,...)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cRuntimeError::cRuntimeError           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classcObject.html">cObject</a> *&nbsp;</td>
          <td class="paramname"> <em>where</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&nbsp;</td>
          <td class="paramname"> <em>...</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
To be called like printf(). 
<p>
The error code is set to eCUSTOM. The 1st arg is the object where the error occurred: its class and object name will be prepended to the message like this: "(cArray)arr". 
</div>
</div><p>
<a class="anchor" name="413646c577852fa82a848ab0187c57c8"></a><!-- doxytag: member="cRuntimeError::cRuntimeError" ref="413646c577852fa82a848ab0187c57c8" args="(const cRuntimeError &amp;e)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">cRuntimeError::cRuntimeError           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classcRuntimeError.html">cRuntimeError</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>e</em>          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
We unfortunately need to copy exception objects when handing them back from an activity(). 
<p>

</div>
</div><p>
<hr><h2>Member Function Documentation</h2>
<a class="anchor" name="6d34e0bf66e83e7c209bb72df3b816f8"></a><!-- doxytag: member="cRuntimeError::dup" ref="6d34e0bf66e83e7c209bb72df3b816f8" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual <a class="el" href="classcRuntimeError.html">cRuntimeError</a>* cRuntimeError::dup           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"> const<code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Virtual copy constructor. 
<p>
We unfortunately need to copy exception objects when handing them back from an activity(). 
<p>Reimplemented from <a class="el" href="classcException.html#7aae9b928c595645f2918a897eb14c2b">cException</a>.</p>

</div>
</div><p>
<hr>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="cexception_8h-source.html">cexception.h</a></ul>
</div>
<hr size="1"><address style="text-align: right;"><small>Generated on Tue Mar 27 11:22:13 2012 for OMNeT++ Simulation Library by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 1.5.5 </small></address>
</body>
</html>
